flytekit/annotated/task.py

```diff
@@ -155,53 +184,82 @@ def name(self) -> str:
         super().__init__(name=task_function.__name__, interface=interface, metadata=metadata, *args, **kwargs)
         self._task_function = task_function
 
-    def execute(self, ctx: FlyteContext,
-                input_literal_map: _literal_models.LiteralMap) -> _literal_models.LiteralMap:
-        """
-        This method translates Flytes native Type system based inputs and dispatches the actual call to the executor
-        TODO We need to figure out for custom types what should be passed into runtime. I feel this should only be inputs,
-            though in case of local execution we should pass the information of the task too.
-        """
-        # Translate the input literals to Python native
-        native_inputs = flytekit_engine.idl_literal_map_to_python_value(ctx, input_literal_map)
-        # TODO maybe we should replace the call of task_function to the actual executor. Thus instead of holding on to
-        #     the function we just hold onto the executor
-        native_outputs = self._task_function(**native_inputs)
-        expected_output_names = list(self.interface.outputs.keys())
-        if len(expected_output_names) == 1:
-            native_outputs_as_map = {expected_output_names[0]: native_outputs}
-        else:
-            # Question: How do you know you're going to enumerate them in the correct order? Even if autonamed, will
-            # output2 come before output100 if there's a hundred outputs? We don't! We'll have to circle back to
-            # the Python task instance and inspect annotations again. Or we change the Python model representation
-            # of the interface to be an ordered dict and we fill it in correctly to begin with.
-            native_outputs_as_map = {expected_output_names[i]: native_outputs[i] for i, _ in
-                                     enumerate(native_outputs)}
+    def execute(self, **kwargs) -> Any:
+        return self._task_function(**kwargs)
 
-        # We manually construct a LiteralMap here because task inputs and outputs actually violate the assumption
-        # built into the IDL that all the values of a literal map are of the same type.
-        outputs_literal_map = _literal_models.LiteralMap(literals={
-            k: flytekit_engine.python_value_to_idl_literal(ctx, v, self.interface.outputs[k].type) for k, v in
-            native_outputs_as_map.items()
-        })
-        print("Outputs!")
-        print(outputs_literal_map)
-        return outputs_literal_map
+    def native_interface(self) -> Interface:
+        return self._native_interface
 
-    def _passthrough_execute(self, **kwargs):
+
+class PysparkFunctionTask(PythonFunctionTask):
+    def __init__(self, task_function: Callable, metadata: _task_model.TaskMetadata, *args, **kwargs):
+        super(PysparkFunctionTask, self).__init__(task_function, metadata,
+                                                  ignore_input_vars=["spark_session", "spark_context"], *args, **kwargs)
```

Comment:
I feel like I'd rather add more stuff to `PysparkFunctionTask` and subclass `Task` directly, instead of adding functionality at the base layer to remove/add inputs.  Or do you foresee other tasks needing this?

Should spark_session and spark_context be valid inputs for data catalog?